packages/datadog-plugin-pg/test/index.spec.js

```diff
@@ -443,6 +442,68 @@ describe('Plugin', () => {
             })
           })
         })
+        it('service should default to tracer service name', done => {
+          tracer
+          agent.use(traces => {
+            expect(traces[0][0]).to.have.property('service', 'test-postgres')
+            done()
+          })
+
+          client.query('SELECT $1::text as message', ['Hello World!'], (err, result) => {
+            if (err) return done(err)
+
+            client.end((err) => {
+              if (err) return done(err)
+            })
+          })
+        })
+      })
+      describe('DBM propagation enabled with full should handle prepared statements', () => {
+        const tracer = require('../../dd-trace')
+
+        before(() => {
+          return agent.load('pg')
+        })
+        beforeEach(done => {
+          pg = require('../../../versions/pg@>=8.0.3').get()
+
+          tracer.init()
+          tracer.use('pg', {
+            dbmPropagationMode: 'full',
+            service: 'post'
+          })
+
+          client = new pg.Client({
+            host: '127.0.0.1',
+            user: 'postgres',
+            password: 'postgres',
+            database: 'postgres'
+          })
+          client.connect(err => done(err))
+        })
+
+        it('prepared statements should be handled', done => {
+          let queryText = ''
+          const query = {
+            text: 'SELECT $1::text as message'
+          }
+          agent.use(traces => {
+            const traceId = traces[0][0].trace_id.toString(16).padStart(32, '0')
+            const spanId = traces[0][0].span_id.toString(16).padStart(16, '0')
+
+            expect(queryText).to.equal(
+              `/*dddbs='post',dde='tester',ddps='test',ddpv='8.4.0',` +
+              `traceparent='00-${traceId}-${spanId}-00'*/ SELECT $1::text as message`)
+          }).then(done, done)
+          client.query(query, ['Hello world!'], (err) => {
```

Comment:
We already support prepared statements, what we didn't support was query config objects (both text and vars as first argument as an object). So that's what we want to test here!

Reply:
Hi @comp615 I have added specific tests for query config objects in [this PR](https://github.com/DataDog/dd-trace-js/pull/3091), please let me know if this addresses the above.